model/task_queue_service_dependency.go

```diff
@@ -412,9 +395,9 @@ func (d *basicCachedDAGDispatcherImpl) FindNextTask(ctx context.Context, spec Ta
 			}
 
 			taskGroupUnit.runningHosts = numHosts
-			d.taskGroups[taskGroupID] = taskGroupUnit
+			d.setTaskGroup(taskGroupUnit, taskGroupID)
```

Comment:
I know this wasn't directly changed, but could we add a comment to this part that we know that this dispatching isn't foolproof, and that it's possible for this to return a task in the task group that will cause it to exceed the task group's max hosts? The next task route has logic to deal with false dispatches that exceed max hosts (and undo it), but the dispatcher itself is only doing it in a best-effort way.